bin/grunt

```diff
@@ -4,41 +4,53 @@
 
 process.title = 'grunt';
 
-// Especially badass external libs.
-var findup = require('findup-sync');
-var resolve = require('resolve').sync;
-
-// Internal libs.
-var options = require('../lib/cli').options;
-var completion = require('../lib/completion');
-var info = require('../lib/info');
-var path = require('path');
-
-var basedir = process.cwd();
-var gruntpath;
-
-// Do stuff based on CLI options.
-if ('completion' in options) {
-  completion.print(options.completion);
-} else if (options.version) {
-  info.version();
-} else if (options.gruntfile) { //Note: if both `gruntfile` and `base` are set, use `gruntfile`
-  basedir = path.resolve(path.dirname(options.gruntfile));
-} else if (options.base) {
-  basedir = path.resolve(options.base);
-}
-
-try {
-  gruntpath = resolve('grunt', {basedir: basedir});
-} catch (ex) {
-  gruntpath = findup('lib/grunt.js');
-  // No grunt install found!
-  if (!gruntpath) {
-    if (options.version) { process.exit(); }
-    if (options.help) { info.help(); }
-    info.fatal('Unable to find local grunt.', 99);
+var Liftoff = require('liftoff');
+var v8flags = require('v8flags');
+var extensions = require('interpret').jsVariants;
+var nopt = require('nopt');
+var gruntOptions = require('grunt-known-options');
+var completion = require('../lib/completion.js');
+var pkg = require('../package.json');
+
+// Parse `gruntOptions` into a form that nopt can handle.
+var aliases = {};
+var known = {};
+
+Object.keys(gruntOptions).forEach(function(key) {
+  var short = gruntOptions[key].short;
+  if (short) {
+    aliases[short] = '--' + key;
   }
+  known[key] = gruntOptions[key].type;
+});
+
+// Parse them and return an options object.
+var options = nopt(known, aliases, process.argv, 2);
+
+if (options.version) {
+  console.log('grunt-cli v' + pkg.version);
```

Comment:
I think you need to exit after this line as it will cause an exception in the line 53. Here is what I got when I run `grunt --version`:

```
$ grunt --version
grunt-cli v1.2.0

assert.js:84
  throw new assert.AssertionError({
  ^
AssertionError: missing path
    at Module.require (module.js:502:3)
    at require (internal/module.js:20:19)
    at Liftoff.<anonymous> (/home/arkni/.local/lib/node_modules/grunt-cli/bin/grunt:53:17)
    at Liftoff.execute (/home/arkni/.local/lib/node_modules/grunt-cli/node_modules/liftoff/index.js:203:12)
    at module.exports (/home/arkni/.local/lib/node_modules/grunt-cli/node_modules/flagged-respawn/index.js:51:3)
    at Liftoff.<anonymous> (/home/arkni/.local/lib/node_modules/grunt-cli/node_modules/liftoff/index.js:195:5)
    at Liftoff.<anonymous> (/home/arkni/.local/lib/node_modules/grunt-cli/node_modules/liftoff/index.js:170:7)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

Reply:
Note that I runned that command in an arbitrary folder which doesn't have `Grunt` installed to its `node_modules`.

Reply:
I think that would be fixed by doing the same as the original version of the CLI: exit if the `modulePath` is not defined and the user requested the version of the CLI.

Reply:
Sweet, fixed. Thanks for the review!